primo-explore/custom/main.js

```diff
@@ -499,6 +499,94 @@ angular.module('myILL', []).component('prmLoansOverviewAfterAppStoreGenerated',
 
 //End My-ill
 
+// Altmetric plugin
+
+app.controller('FullViewAfterController', ['angularLoad', '$http', '$scope', '$element', '$timeout', '$window', function (angularLoad, $http, $scope, $element, $timeout, $window) {
+  var vm = this;
+  this.$http = $http;
+  this.$element = $element;
+  this.$scope = $scope;
+  this.$window = $window;
+  vm.$onInit = function () //wait for all the bindings to be initialised
+  {
+    vm.parentElement = this.$element.parent()[0]; //the prm-full-view container
+    try {
+        vm.doi = vm.parentCtrl.item.pnx.addata.doi[0] || '';
+    } catch (e) {
+      return;
+    }
+
+    if (vm.doi) {
+      //If we've got a doi to work with check whether altmetrics has data for it.
+      //If so, make our div visible and create a new Altmetrics service
+      $timeout(function () {
+        $http.get('https://api.altmetric.com/v1/doi/' + vm.doi).then(function () {
+            try {
+              //Get the altmetrics widget
+              angularLoad.loadScript('https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js?' + Date.now()).then(function () {});
+              //Create our new Primo service
+              var altmetricsSection = {
+                scrollId: "altmetrics",
+                serviceName: "altmetrics",
+                title: "brief.results.tabs.Altmetrics"
+              };
+              vm.parentCtrl.services.splice(vm.parentCtrl.services.length, 0, altmetricsSection);
+            } catch (e) {
+              console.log(e);
+            }
+        }).catch(function (e) {
+          return;
+        });
+        }, 3000);
+  }
+
+
+  //move the altmetrics widget into the new Altmetrics service section
+  var unbindWatcher = this.$scope.$watch(function () {
+    return vm.parentElement.querySelector('h4[translate="brief.results.tabs.Altmetrics"]');
+  }, function (newVal, oldVal) {
+        if (newVal) {
+          //Get the section body associated with the value we're watching
+          let altContainer = newVal.parentElement.parentElement.parentElement.parentElement.children[1];
```

Comment:
Is there any other way to do this? Calling `parentElement` multiple times seems kind of fragile.

Reply:
Oh wow, I did not even see this. It does look quite fragile, so I will take a quick look to see if there is a better way. Good eye.